cpp/src/gandiva/precompiled/time.cc

```diff
@@ -563,4 +563,108 @@ date64 castDATE_utf8(int64_t context, const char* input, int32 length) {
       .time_since_epoch()
       .count();
 }
+
+/*
+ * Input consists of mandatory and optional fields.
+ * Mandatory fields are year, month and day.
+ * Optional fields are time, displacement and zone.
+ * Format is <year-month-day>[ hours:minutes:seconds][.millis][ displacement|zone]
+ *
+ * ts_fields[0] = year
+ * ts_fields[1] = month
+ * ts_fields[2] = day
+ * ts_fields[3] = hours
+ * ts_fields[4] = minutes
+ * ts_fields[5] = seconds
+ * ts_fields[6] = sub_seconds
+ * ts_fields[7] = displacement_hours
+ * ts_fields[8] = displacement_minutes
+ */
+timestamp castTIMESTAMP_utf8(int64_t context, const char* input, int32 length) {
+  int ts_fields[9] = {0, 0, 0, 0, 0, 0, 0, 0, 0};
```

Comment:
can we please use enums for the fields ?

Reply:
done